frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -1,14 +1,59 @@
 import React, { useContext, useState, useEffect } from 'react'
-import cx from 'classnames'
 import { updateAnnouncementVersion } from 'services/settings.service'
 
-import { PrimaryButton, ErrorBlock } from 'components/common'
 import { ModalContext } from 'contexts/modal.context'
 
-import styles from './AnnouncementModal.module.scss'
 import { i18n } from 'locales'
-import { ANNOUNCEMENT } from 'config'
-import { OutboundLink } from 'react-ga'
+import { ANNOUNCEMENT, getAnnouncementVersion } from 'config'
+import { ErrorBlock } from 'components/common'
+import GraphicAnnouncementModal from './GraphicAnnouncementModal'
+import VideoAnnouncementModal from './VideoAnnouncementModal'
+import styles from './AnnouncementModal.module.scss'
+
+export type Translation = string | undefined
+
+export type AnnouncementModalProps = {
```

Comment:
Rather than declaring every property with the type `string | undefined`, perhaps you want to consider making some properties compulsory and others optional. This way you can encode information about the domain in the type and make it more meaningful.

For example for the type below, it's clear that `title`, `primaryButtonUrl` and `primaryButtonText` are compulsory while `medialUrl` is not:
```ts
export type AnnouncementModalProps = {
    title: string
    primaryButtonUrl: string
    primaryButtonText: string

    mediaUrl?: string
    ...
}
```

Reply:
Thanks, I've done as advised.